src/main/java/com/dev/cinema/dao/impl/GenericDaoImpl.java

```diff
@@ -2,20 +2,30 @@
 
 import com.dev.cinema.dao.GenericDao;
 import com.dev.cinema.exeption.DataProcessingExeption;
-import com.dev.cinema.util.HibernateUtil;
 import lombok.extern.log4j.Log4j;
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
 
 @Log4j
+@Repository
```

Comment:
should we use Repository here?

Reply:
We have this class as part of other classes, it is not individual so we no need it here.